### PR TITLE
Fixed dnsmasq disable code under port 53 check

### PR DESCRIPTION
**Pull request by RamSet**
Signed-off-by: RamSet <RamSet@gmail.com>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
*A detailed description, screenshots (if necessary), as well as links to any relevant GitHub issues*

fixed failure on install when dnsmasq was started by init.d scrip

**How does this PR accomplish the above?:**
*A detailed description (such as a changelog) and screenshots (if necessary) of the implemented fix*

check for systemctl or init.d via the built in functions.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1505,19 +1505,24 @@ disable_dnsmasq() {
...
             echo -e "  ${EXCL} dnsmasq still active, this is most likely due to the fact that ${COL_LIGHT_RED}dnsmasq"
-            echo -e "    was loaded via a non convetional method. This might cause future conflicts with FTLDNS${COL_NC}"
-            ${SUDO} pkill dnsmasq
-            echo -e "  ${TICK} dnsmasq process killed"
+            echo -e "    was loaded via an init.d service.${COL_NC}"
+            # Stopping dnsmasq
+            echo -e "  ${TICK} Stopping dnsmasq via service command"
+            ${SUDO} service dnsmasq stop > /dev/null 2>&1
+            # Disabling dnsmasq
+            echo -e "  ${TICK} Disabling dnsmasq via update-rc.d command"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: These should be info instead of tick, since they haven't completed yet and are just info logs.

**Comment by Mcat12** [comment]
Note that this also adds the port check back (reverted in #2302)

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1398,7 +1486,45 @@ disable_resolved_stublistener() {
...
+            # Make a backup of the original /etc/NetworkManager/NetworkManager.conf
+            # (This will need to be restored on uninstallation)
+            ${SUDO} sed -r -i.orig '/#?dns=dnsmasq/d' /etc/NetworkManager/NetworkManager.conf
+            echo -e "  ${TICK} Restarting Network manager"
+            ${SUDO} systemctl reload-or-restart NetworkManager
+        else
+            # Disabling dnsmasq
+            disable_service dnsmasq
+            # Stopping dnsmasq
+            stop_service dnsmasq
```
Comment (Mcat12) [inline automated install/basic-install.sh]: The service should be stopped before disabling.

**Comment by dschaper** [comment]
Force-push reset audit for code changes, unable to determine if pushed changes resolve requested changes.

**Comment by DL6ER** [comment]
Merge conflict - removing v4.1 milestone

**Comment by dschaper** [comment]
This is kind of outdated at this point. @RamSet Did you want to give this a new go on a new branch/PR?

**Comment by RamSet** [comment]
Sure will 😉 on tuesday. Thermal on monday :)

**Comment by RamSet** [comment]
Will review under new PR

**Pull request closed without merging**